File: module_ui/src/main/java/com/module/ui/adapter/MainAdapter.java
package com.module.ui.adapter;

import android.app.Activity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.module.ui.R;
import com.module.ui.bean.UIItem;

import java.util.ArrayList;
import java.util.List;

import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by wangwei on 2018/9/27.
 */

public class MainAdapter extends RecyclerView.Adapter<MainAdapter.MyViewHolder> {
    private Activity mContext;
    private final LayoutInflater mLayoutInflater;

    public ArrayList<UIItem> mDatas = new ArrayList<>();

    public void addItem(UIItem item) {
        mDatas.add(item);
    }

    public UIItem getItem(int index) {
      return   mDatas.get(index);
    }
    public void addAll(List<UIItem> data) {
        mDatas.addAll(data);
    }


    public interface OnItemClickLitener {
        void onItemClick(int index);
    }

    private MainAdapter.OnItemClickLitener mOnItemClickLitener;

    public void setOnItemClickLitener(MainAdapter.OnItemClickLitener mOnItemClickLitener) {
        this.mOnItemClickLitener = mOnItemClickLitener;
    }

    public MainAdapter(Activity context) {
        this.mContext = context;
        mLayoutInflater = LayoutInflater.from(context);

    }


    @Override
    public MainAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent, int i) {
        View inflate = mLayoutInflater.inflate(R.layout.main_item, parent, false);
        MainAdapter.MyViewHolder masterViewHolder = new MainAdapter.MyViewHolder(inflate);
        return masterViewHolder;
    }

    @Override
    public void onBindViewHolder(final MainAdapter.MyViewHolder holder, final int position) {
        final String oj = mDatas.get(position).getTitle();
        upDateHolderView(holder, oj, position);
        // 如果设置了回调，则设置点击事件
        if (mOnItemClickLitener != null) {
            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mOnItemClickLitener.onItemClick(position);
                }
            });
        }
        holder.itemView.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                Log.w("TAG", "---" + event.getAction());
                return false;
            }
        });
    }

    private void upDateHolderView(MainAdapter.MyViewHolder holder, String oj, int position) {
        holder.item_tv.setText(oj);
    }

    @Override
    public int getItemCount() {
        return mDatas.size();

    }

    class MyViewHolder extends RecyclerView.ViewHolder {
        private View itemView;
        public TextView item_tv;

        public MyViewHolder(View itemView) {
            super(itemView);
            this.itemView = itemView;
            item_tv = itemView.findViewById(R.id.item_tv);
        }

    }

}

File: module_ui/src/main/java/com/module/ui/adapter/MainAdapter.kt
package com.module.ui.adapter

import android.app.Activity
import android.util.Log
import android.view.LayoutInflater
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.module.ui.R
import com.module.ui.bean.UIItem

import java.util.ArrayList

import androidx.recyclerview.widget.RecyclerView
import com.module.ui.activity.UIActivity

/**
 * Created by wangwei on 2018/9/27.
 */

class MainAdapter(val mContext: Activity) : RecyclerView.Adapter<MainAdapter.MyViewHolder>() {
    var mData = ArrayList<UIItem>()
    private var mOnItemClickListener: OnItemClickListener? = null

    //初始化数据
    init {

        Log.w("TAG", "---init${UIActivity.start("4566")}" )
        Log.w("TAG", "---init" + UIActivity.Companion.name)

    }
    fun addItem(item: UIItem) {
        mData.add(item)
    }

    fun getItem(index: Int): UIItem {
        return mData[index]
    }

    fun addAll(data: List<UIItem>) {
        mData.addAll(data)
    }


    interface OnItemClickListener {
        fun onItemClick(index: Int)
    }

    fun setOnItemClickListener(lis: OnItemClickListener) {
        this.mOnItemClickListener = lis
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val inflate = LayoutInflater.from(mContext).inflate(R.layout.main_item, parent, false)
        return MyViewHolder(inflate)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val oj = mData[position].title
        upDateHolderView(holder, oj, position)
        // 如果设置了回调，则设置点击事件
        if (mOnItemClickListener != null) {
            holder.itemView.setOnClickListener { mOnItemClickListener!!.onItemClick(position) }
        }
    }

    private fun upDateHolderView(holder: MainAdapter.MyViewHolder, oj: String, position: Int) {
        holder.itemTv.text = oj
    }

    override fun getItemCount(): Int {
        return mData.size
    }

     inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var itemTv: TextView = itemView.findViewById(R.id.item_tv)
       //  var  itemView:View=view


     }

}



